### PR TITLE
[main] bmake: add patch to disable broken tests

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

bmake has some broken unit tests related to localtime which fail unexpectedly sometimes. 
The latest release, 20220303, has no changes to these broken tests. 
The bmake tests currently randomly fail our pipelines every so often, costing us productivity. 
The patch disables three tests in varmod-localtime.mk, and a bug item has been made for them here: [38644519](https://microsoft.visualstudio.com/OS/_workitems/edit/38644519)

###### Change Log  

- Add patch remove-inconsistent-time-tests.patch, which disables unreliably failing tests

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Mock build

### CODE REVIEW

**Review by MaxBrodeurUrbas — APPROVED** [review]
Thanks for these changes, looks great.

File: SPECS/bmake/bmake.spec [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 Name:          bmake
 Version:       20211221
-Release:       1%{?dist}
+Release:       2%{?dist}
 License:       BSD
 Vendor:        Microsoft Corporation
 Distribution:  Mariner
 URL:           https://ftp.netbsd.org/pub/NetBSD/misc/sjg/
 Source0:       %{url}/bmake-%{version}.tar.gz
+Patch0:        disable-time-tests.patch
```
Comment (PawelWMS) [inline SPECS/bmake/bmake.spec:10]: Please also add a comment with an explanation for this patch and a link to the bug, which forces us to apply it. This way in the future it's going to be easier to find it and check, if the issue is still unresolved.

**Review by PawelWMS — APPROVED** [review]
Looks good, just a small request to add a comment closer to the patch.

**Review by christopherco — APPROVED** [review]

File: SPECS/bmake/bmake.spec [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+Release:       2%{?dist}
 License:       BSD
 Vendor:        Microsoft Corporation
 Distribution:  Mariner
 URL:           https://ftp.netbsd.org/pub/NetBSD/misc/sjg/
 Source0:       %{url}/bmake-%{version}.tar.gz
+# Certain tests in varmod-localtime.mk will inconsistently fail
+# on pipeline machines. Disabling with this patch for now, and
+# tracking this bug in workitem 38644519
+Patch0:        remove-inconsistent-time-tests.patch
```
Comment (jslobodzian) [inline SPECS/bmake/bmake.spec:13]: Before we take this, do we understand why the test is failing?  Is there a fix or patch upstream? Is there anything simple we can do to make the test reliable?

**Pull request merged**